base/src/main/java/uk/ac/ebi/atlas/experimentimport/analyticsindex/conditions/ConditionsCollector.java

```diff
@@ -36,22 +34,33 @@ public ConditionsCollector(EfoLookupService efoLookupService) {
                         experiment.getExperimentDesign().getSampleCharacteristicsValues(assayAccession).values())
                 .flatMap(Collection::stream)
                 .collect(toSet());
-        }
+    }
 
     // Visits all assays in each factor and builds-collects the conditions in a stream
     public Stream<Condition> getConditions(BaselineExperiment experiment) {
+        ImmutableSetMultimap<String, String> assayAccession2OntologyTerms =
```

Comment:
This should probably go in a separate (private) function to avoid code duplication: lines 60-63 seem identical to this block, unless I'm missing something...

Reply:
This is the usual compromise on DRY vs “flow” (for lack of a better word). In this case I thought that adding a level of indirection wouldn’t add value to code readability, but now I’m not so sure. I’ve added a commit to change this. Let me know if you think this reads better.